torch/jit/__init__.py

```diff
@@ -354,22 +355,29 @@ def __getattr__(self, attr):
         return self.module._get_method(attr)
 
 
-def _script_graph(fn, _frames_up=0):
+def script(fn, optimize=True, _frames_up=0):
     rcb = createResolutionCallback(_frames_up + 1)
     ast = get_jit_ast(fn)
-    return _jit_script_compile(ast, rcb)
-
-
-def script(fn, optimize=True, _frames_up=0):
-    graph = _script_graph(fn, _frames_up=_frames_up + 1)
+    schema = annotations.get_signature(fn)
+    if schema:
+        typed_def = torch._C._pack_typed_def(ast, schema[0], schema[1])
```

Comment:
Can't we expose `FunctionSchema` into Python and use the regular constructor of `TypedDef` instead? There's no reason to expand the API surface

Reply:
That's what I originally had but pybind bindings for `at::optional` are broken, which prevents us from exposing both the `TypedDef` and `Argument` constructors to Python directly. Without falling into the rabbit hole of fixing weird pybind stuff, this is the workaround